### ISSUE TITLE
MembersToInclude failure with List property

### ISSUE THREAD

**Issue by Modestas** [GregFinzer/Compare-Net-Objects#190]
Project info
```
<TargetFramework>netcoreapp2.2</TargetFramework>
...
<PackageReference Include="CompareNETObjects" Version="4.64.0" />
```

Classes
```
public class Parent {
    public List<Child> Children { get; set; }
}
public class Child {
    public bool BoolA { get; set; }
    public bool BoolB { get; set; }
}
...
Parent a = new Parent {
    Children = new List<Child> 
    {
        new Child {
	    BoolA = true,
	    BoolB = false
	},
	new Child {
	    BoolA = false,
	    BoolB = true
	}
    }
}
Parent b = new Parent {
    Children = new List<Child> 
    {
	new Child {
	    BoolA = false,
	    BoolB = true
	},
	new Child {
	    BoolA = true,
	    BoolB = false
	}
    }
}
```
AreEqual = false (as expected):
```
ComparisonConfig config = new ComparisonConfig()
CompareLogic compareLogic = new CompareLogic(config);
ComparisonResult result = _compareLogic.Compare(a, b);
//result.AreEqual = false (as expected)
```
AreEqual = true (bug):
```
ComparisonConfig config = new ComparisonConfig 
{
    MembersToInclude = new List<string>
    {
        "Children"
    }
};
CompareLogic compareLogic = new CompareLogic(config);
ComparisonResult result = _compareLogic.Compare(a, b);
//result.AreEqual = true (bug)
```

**Comment by GregFinzer** [comment]
This is by design.  None of the child properties are included so they are not compared.  You will also need to add Parent, Child, BoolA, and BoolB

**Comment by aurimasMisius** [comment]
The current way does not work if I had a property named the same (e.g. Id) in both parent and child and would like to compare child's Id, but not parent's. Is there a way to achive that? If I did MembersToInclude = new List<string>
    {
        "Id"
    }
that would compare Id from both parent and child, right?

**Comment by GregFinzer** [comment]
You should be able to specify Child.Id

**Comment by aurimasMisius** [comment]
I tried it, but it still not work. Could it be due to child being in a list? The setup I have is as Modestas posted before and the config setup is like this:
`ComparisonConfig config = new ComparisonConfig
            {
                MembersToInclude = new List<string>
                {
                    "Child.BoolB",
                    "Child[0].BoolB",
                    "Children.BoolB",
                    "Children[0].BoolB"
                }
            };
            _compareLogic = new CompareLogic(config);`
I believe this way if at least one of the strings was correct way to adress a Child's property the comparartor should return false, but when I run it always returns true. I did work around and compared explicitly as there is only 2 properties, but would like to find a proper way to do it.
